format/tzx/tzx.go

```diff
@@ -0,0 +1,655 @@
+			length := d.FieldU24("length") // Length of data that follows
+
+			// Data as in .TAP files
+			d.FieldRawLen("data", int64(length*8))
+		},
+
+		// ID: 12h (18d) | Pure Tone
+		// This will produce a tone which is basically the same as the pilot
+		// tone in 10h and 11h blocks.
+		0x12: func(d *decode.D) {
+			d.FieldU16("pulse_length") // Length of one pulse in T-states
+			d.FieldU16("pulse_count")  // Number of pulses
+		},
+
+		// ID: 13h (19d) | Sequence of Pulses
+		// This will produce N pulses, each having its own timing. Up to 255
+		// pulses can be stored in this block.
+		0x13: func(d *decode.D) {
+			count := d.FieldU8("pulse_count")
+			d.FieldArray("pulses", func(d *decode.D) {
+				for i := uint64(0); i < count; i++ {
+					d.FieldU16("pulse")
+				}
+			})
+		},
+
+		// ID: 14h (20d) | Pure Data
+		// This is the same as in the turbo loading data block, except that it
+		// has no pilot or sync pulses.
+		0x14: func(d *decode.D) {
+			d.FieldU16("bit0_pulse")       // Length of ZERO bit pulse
+			d.FieldU16("bit1_pulse")       // Length of ONE bit pulse
+			d.FieldU8("used_bits")         // Used bits in last byte
+			d.FieldU16("pause")            // Pause after this block (ms.)
+			length := d.FieldU24("length") // Length of data that follows
+
+			// Data as in .TAP files
+			d.FieldRawLen("data", int64(length*8))
+		},
+
+		// ID: 15h (21d) | Direct Recording
+		// This block is used for tapes which have some parts in a format such
+		// that the turbo loader block cannot be used. This is not like a VOC
+		// file since the information is much more compact. Each sample value
+		// is represented by one bit only (0 for low, 1 for high) which means
+		// that the block will be at most 1/8 the size of the equivalent VOC.
+		// The preferred sampling frequencies are 22050 or 44100 Hz
+		// (158 or 79 T-states/sample).
+		0x15: func(d *decode.D) {
+			d.FieldU16("t_states")                 // Number of T-states per sample (bit of data)
+			d.FieldU16("pause")                    // Pause after this block in milliseconds (ms.)
+			d.FieldU8("used_bits")                 // Used bits (samples) in last byte of data (1-8)
+			length := d.FieldU24("length")         // Length of data that follows
+			d.FieldRawLen("data", int64(length*8)) // Samples data. Each bit represents a state on the EAR port
+		},
+
+		// ID: 18h (24d) | CSW Recording
+		// This block contains a sequence of raw pulses encoded in CSW format
+		// v2 (Compressed Square Wave).
+		0x18: func(d *decode.D) {
+			length := d.FieldU32("length") // Block length (without these four bytes)
+
+			// NOTE: remove these next 4 fields from the length so
+			// the data size is calculated correctly
+			length -= 2 + 3 + 1 + 4
+
+			// Pause after this block (in ms)
+			d.FieldU16("pause")
+			// Sampling rate
+			d.FieldU24("sample_rate")
+			// Compression type
+			d.FieldU8("compression_type", scalar.UintMapSymStr{0x01: "RLE", 0x02: "Z-RLE"})
```

Comment:
I've usually tried to keep symbol mapped strings to be jq-ish, that is snake_case, but if the format spec etc is very clear about the naming then keep it think.